### ISSUE TITLE
Collaboration

### ISSUE THREAD

**Issue by benjchristensen** [electronicarts/orbit#7]
Hi folks, your new project looks great and possibly like something I (at Netflix) could benefit from. We've been exploring Project Orleans and hadn't found something in Java and were considering our own implementation. We need an AP (as in CAP theorem) system that can provide stateful pub/sub with 100M+ topics/queues/actors and Orbit looks like it may fit the bill. 

Can you provide information on your roadmap? If we have items we'd like to suggest changes for to enable adapting to our environment are you open to discussing them as issues?

If we find this fits our needs we are interested in different persistence providers (beyond Mongo) and discovery mechanisms (not JGroups) for example.

**Comment by JoeHegarty** [comment]
Hi Ben, great to hear from you. 

I don't have a clear roadmap that I can share right now but I can call out some of the things that I think we'll be looking at in the short-term:
* Support special placement strategies for actors (Prefer local, limited number of actors of a type per node etc)
* Further abstraction of implementation details (Not relying directly on JGroups, Infinispan etc, exposing them using the existing Provider system)
* Offer runtime byte-code generation as well as compile time code generation
* Offer a generic system for aggregating large volumes of data without causing a bottleneck (Analytics, BI etc)
I'd like to put together a more firm roadmap in the coming weeks now that we've got the project out in the open.

We're definitely open to getting input on what others interested in the framework would like to see. We're also open to taking changes from others, especially where those changes make the framework more generic and allow people to replace aspects that don't suit them.

Storage and Lifetime providers can already be implemented as separate modules. Our default storage provider is MongoDB, and earlier today we accepted a PostgreSQL provider. We're definitely open to taking back more providers.

It would be great if we could collaborate and make the framework better for everyone.

Thanks,
Joe
